src/map/pc.cpp

```diff
@@ -6066,22 +6066,18 @@ enum e_setpos pc_setpos(struct map_session_data* sd, unsigned short mapindex, in
 		if (sc && sc->count) { // Cancel some map related stuff.
 			if (sc->data[SC_JAILED])
 				return SETPOS_MAPINDEX; //You may not get out!
-			status_change_end(&sd->bl, SC_BOSSMAPINFO, INVALID_TIMER);
-			status_change_end(&sd->bl, SC_WARM, INVALID_TIMER);
-			status_change_end(&sd->bl, SC_SUN_COMFORT, INVALID_TIMER);
-			status_change_end(&sd->bl, SC_MOON_COMFORT, INVALID_TIMER);
-			status_change_end(&sd->bl, SC_STAR_COMFORT, INVALID_TIMER);
-			status_change_end(&sd->bl, SC_MIRACLE, INVALID_TIMER);
-			if (sc->data[SC_KNOWLEDGE]) {
-				struct status_change_entry *sce = sc->data[SC_KNOWLEDGE];
+			for (const auto &it : status_db) {
+				sc_type status = static_cast<sc_type>(it.first);
+
+				if (sd->sc.data[status] && it.second->flag[SCF_REM_ON_MAPWARP])
+					status_change_end(&sd->bl, status, INVALID_TIMER);
+			}
+			if (sd->sc.data[SC_KNOWLEDGE]) {
```

Comment:
Flag "restart on warp"?
Or what is this doing here? Am I misinterpreting it?

Reply:
From reading the block below, the timer restarts for this SC when a player moves to another map and doesn’t actually end so this doesn’t fit the SCF_REM_ON_MAPWARP case.